### PR TITLE
ADR-016: Add protocol Version to NodeInfo

### PR DESCRIPTION
**Pull request by ebuchman**
More from https://github.com/tendermint/tendermint/issues/2468

* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: p2p/node_info.go [hunk 1]
```diff
@@ -124,36 +137,40 @@ func (info DefaultNodeInfo) ValidateBasic() error {
...
+
+	// Validate Other.
+	other := info.Other
+	txIndex := other.TxIndex
+	switch txIndex {
+	case "", "on", "off":
+	default:
+		return fmt.Errorf("info.Other.TxIndex should be either 'on' or 'off', got '%v'", txIndex)
+	}
+	// XXX: Should we be more strict about address formats?
```
Comment (Liamsi) [inline p2p/node_info.go]: Are they always IP4 addresses?

Comment (ebuchman) [reply p2p/node_info.go]: No actually, in theory they could be IPv6. We don't actually do anything with this RPCAddress though, just display it over RPC, so it doesn't matter too much here ...

File: p2p/node_info.go [hunk 2]
```diff
@@ -124,36 +137,40 @@ func (info DefaultNodeInfo) ValidateBasic() error {
...
+		return fmt.Errorf("info.Moniker must be valid non-empty ASCII text without tabs, but got %v", info.Moniker)
+	}
+
+	// Validate Other.
+	other := info.Other
+	txIndex := other.TxIndex
+	switch txIndex {
+	case "", "on", "off":
+	default:
+		return fmt.Errorf("info.Other.TxIndex should be either 'on' or 'off', got '%v'", txIndex)
```
Comment (melekes) [inline p2p/node_info.go]: 'on' or 'off' or empty string

Comment (ebuchman) [reply p2p/node_info.go]: Thanks - bb173bcae29e1aa76cff0bf842822318d87fe77b

**Pull request merged**